students/davidkan/session03/mailroom.py

```diff
@@ -0,0 +1,176 @@
+                                    "\n> "))
+    except (ValueError, UnboundLocalError) as e:
+        print("Please type only integers.")
+
+    return donation_amount
+
+
+def add_donation_to_user(donor_name_index, donation_amount):
+    """
+    Function will add the amount given to the donation history of the selected user
+    :param donor_name_index: the index location of the donor
+    :param donation_amount: the amount the donor donated
+    :return: None
+    """
+    # Append the donation amount to the donor's name in the donor list data structure
+    donors[donor_name_index][1].append(donation_amount)
+
+
+def print_email_to_terminal(full_name, donation_amount):
+    """
+    Function will print the thank you email to the donor on the terminal
+    :param full_name: username in lowercase
+    :param donation_amount: the amount the donor donated
+    :return: None
+    """
+    print()
+    # Finally, use string formatting to compose an email thanking the donor for their generous donation.
+    # Print the email to the terminal and return to the original prompt
+    print("Dear {0},\n\n"
+          "Thank you very much for your recent donation of ${1} to the local mailing room at our local charity.\n"
+          "I would like to now invite you to subscribe to our mailing room group, so that you can learn more\n"
+          "about how your support is helping us. Again, thank you very much for your support, we greatly\n"
+          "appreciate it.\n\n"
+          "Best regards,\n\n"
+          "Mailing Group Association\n".format(full_name.title(), donation_amount))
+
+
+def send_thank_you():
+    """
+    Function will send a thank you email and print it to the terminal
+    :return: None
+    """
+    while True:
+        resuts = input("\nFull Name:\n"
+                       "\n> ")
+        full_name = resuts.lower()
+
+        if 'list' in full_name:
+            show_donor_list()
+            continue
+
+        if full_name not in [row[0] for row in donors]:
+            add_donor(full_name)
+
+        donor_name_index = get_donors_index(full_name)
+        donation_amount = get_donation_amount()
+        add_donation_to_user(donor_name_index, donation_amount)
+        print_email_to_terminal(full_name, donation_amount)
+
+        break
+
+
+def create_report():
+    """
+    Function will print out the report in a tabular formatted view
+    :return: None
+    """
+    # Create header for report
+    print("\n{:<10} | {:>10} | {:>10} | {:>10}".format('Donor Name', 'Total Given', 'Num Gift', 'Average Gift'))
+    print("{}".format(''.join(['-' * 52])))
+
+    # Create list that includes donor name, total given, number of gifts, and average gift as values in each row
+    donors_list = [(row[0], sum(row[1]), len(row[1]), int(sum(row[1]) / len(row[1]))) for row in donors]
+
+    # Sort total given by descending order
+    ordered_donors_list = sorted(donors_list, key=lambda x: (x[1],x[1]), reverse=True)
```

Comment:
very nice use of sorted() and lambda for the sort key.

but why (x[1],x[1]) ?? that's going to sort the exact same as x[1]